src/parser/ion_1_0/text/mod.rs

```diff
@@ -0,0 +1,2614 @@
+            "sexp" => ion::Data::Sexp(None),
+            _ => unreachable!("condition covers all possible parsed results"),
+        }
+    };
+
+    Ok((i, data))
+}
+
+/// TODO: add field names parsed as strings to the symbol table.
+///
+/// field_name
+///     : symbol
+///     | SHORT_QUOTED_STRING
+///     | (ws* LONG_QUOTED_STRING)+
+///     ;
+fn take_field_name(table: Table) -> impl Fn(&str) -> IonResult<&str, SymbolToken> {
+    move |i: &str| {
+        alt((
+            take_symbol(table.clone()),
+            map(take_short_quoted_string, |text| SymbolToken::Known { text }),
+            map(many1(preceded(eat_ws, take_long_quoted_string)), |v| {
+                SymbolToken::Known { text: v.concat() }
+            }),
+        ))(i)
+    }
+}
+
+/// quoted_text
+///     : QUOTED_SYMBOL
+///     | SHORT_QUOTED_STRING
+///     | (ws* LONG_QUOTED_STRING)+
+///     ;
+fn take_quoted_text(i: &str) -> IonResult<&str, ion::Data> {
+    alt((
+        map(take_quoted_symbol, |text| {
+            ion::Data::Symbol(Some(SymbolToken::Known { text }))
+        }),
+        map(take_short_quoted_string, |s| ion::Data::String(Some(s))),
+        map(
+            map(many1(preceded(eat_ws, take_long_quoted_string)), |v| {
+                v.concat()
+            }),
+            |s| ion::Data::String(Some(s)),
+        ),
+    ))(i)
+}
+
+fn get_or_make_symbol(table: Table, text: String) -> Result<SymbolToken, SymbolError> {
+    if let Ok((_, s)) = take_sid(&text) {
+        let sid = s
+            .parse::<usize>()
+            .map_err(|_| SymbolError::InvalidSid(s.to_string()))?;
+        match table.borrow().lookup_sid(sid) {
+            Ok(token) => Ok(token),
+            Err(e) => Err(e),
+        }
+    } else {
+        make_symbol(table, text)
+    }
+}
+
+fn make_symbol(table: Table, text: String) -> Result<SymbolToken, SymbolError> {
+    let token = SymbolToken::Known { text };
+    table.borrow_mut().add_symbol(&token);
+    Ok(token)
+}
+
+/// Note: IDENTIFIER_SYMBOL covers all cases provided by TYPE.
+///
+/// symbol
+///     : IDENTIFIER_SYMBOL
+///     // note that this is because we recognize the type names for null
+///     // they are ordinary symbols on their own
+///     | TYPE
+///     | QUOTED_SYMBOL
+///     ;
+fn take_symbol(table: Table) -> impl Fn(&str) -> IonResult<&str, SymbolToken> {
+    move |i: &str| {
+        let (i, result) = alt((
+            map(take_identifier_symbol, |s| {
+                get_or_make_symbol(table.clone(), s.to_string())
+            }),
+            map(take_quoted_symbol, |s| make_symbol(table.clone(), s)),
+        ))(i)?;
+
+        match result {
+            Ok(token) => Ok((i, token)),
+            Err(e) => Err(Err::Failure(IonError::from_symbol_error(i, e))),
+        }
+    }
+}
+
+fn take_sid(i: &str) -> IonResult<&str, &str> {
+    preceded(char('$'), take_while1(is_dec_digit))(i)
+}
+
+fn eat_ws(i: &str) -> IonResult<&str, &str> {
+    recognize(many0(ws))(i)
+}
```

Comment:
The `many0` makes me think this should be named `eat_opt_ws`

Reply:
Good call. Originally I kept it called `take_ws` since it was the parser analogue for the `ws` ANTLR object, but they diverged when I got sick of adding `many0(..)` to every invocation site.